### PR TITLE
Tweaks to vote margin requirement

### PR DESCRIPTION
**Pull request by Legomenon-gh**
I'm not super happy with this implementation (it's a bit ugly), but it's marginally better than what was there so it might be worth a pull request (it makes it easier to make limited changes to poll thresholds in the future). Feel free to reject it though, I'd prefer a broader refactor but can't justify the risk vs reward ratio.

https://github.com/ZeroK-RTS/Zero-K-Infrastructure/commit/9f659704e25239216b952876ba9858188e595f53 is based on forum comments (I personally have no stakes on large team games). I'm happy to take it out if making resign universally harder is considered a good change.

Addresses https://github.com/ZeroK-RTS/Zero-K-Infrastructure/issues/2808

### CODE REVIEW

**Comment by Legomenon-gh** [comment]
https://github.com/ZeroK-RTS/Zero-K-Infrastructure/commit/9f659704e25239216b952876ba9858188e595f53 should be irrelevant pretty soon since the game will likely be handling resign votes instead of the server.

Not sure if the first commit is worth it now, feel free to close it

File: ZkLobbyServer/autohost/Commands/CmdStart.cs [hunk 1]
```diff
@@ -54,6 +54,17 @@ public override async Task ExecuteArmed(ServerBattle battle, Say e)
             }
 
         }
+        public override int GetPollWinMargin(ServerBattle battle, int numVoters)
+        {
+            // Require one more vote to start a game with uneven teams so at least one player in the smaller
+            // team needs to agree to start the game. This is particularly relevant for the 2v1 case.
+            if (battle.Mode == PlasmaShared.AutohostMode.Teams) {
+                return 1 + numVoters % 2;
```
Comment (sprunk) [inline ZkLobbyServer/autohost/Commands/CmdStart.cs:62]: The 1 is probably meant to be "whatever the base margin is"?
```suggestion
                return base.GetPollWinMargin(battle, numVoters) + numVoters % 2;
```

Comment (Legomenon-gh) [reply ZkLobbyServer/autohost/Commands/CmdStart.cs:62]: Yup let's be consistent

**Review by sprunk — APPROVED** [review]
looks good

**Pull request merged**